data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/accumulator/BufferTypeOptions.java

```diff
@@ -5,13 +5,34 @@
 
 package org.opensearch.dataprepper.plugins.sink.accumulator;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Collectors;
+
 /**
  * Defines all the buffer types enumerations.
  */
 public enum BufferTypeOptions {
 
-    INMEMORY,
-    LOCALFILE
+    INMEMORY("in_memory", new InMemoryBuffer());
+
+    private final String option;
+    private final Buffer bufferType;
+    private static final Map<String, BufferTypeOptions> OPTIONS_MAP = Arrays.stream(BufferTypeOptions.values())
+            .collect(Collectors.toMap(value -> value.option, value -> value));
+
+    BufferTypeOptions(final String option, final Buffer bufferType) {
+        this.option = option.toLowerCase();
+        this.bufferType = bufferType;
+    }
+
+    public Buffer getBufferType() {
+        return bufferType;
+    }
 
-    //TODO add buffer type options and remaining functionality
+    @JsonCreator
+    static BufferTypeOptions fromOptionValue(final String option) {
+        return OPTIONS_MAP.get(option.toLowerCase());
```

Comment:
Let's remove the `toLowerCase()` here. We should only allow users to specify the exact string here.

Reply:
Resolved.